mappings/bnfdmd/data_downloader.py

```diff
@@ -0,0 +1,129 @@
+import glob
+import os
+import re
+from datetime import datetime
+from pathlib import Path
+from urllib.parse import unquote, urljoin, urlparse
+
+import requests
+import structlog
+from bs4 import BeautifulSoup
```

Comment:
thought: Having just been looking at lots of dependencies, BeautifulSoup introduces an additional two. It's a *little* bit of a smell, because we already have lxml as a dependency, which we could use instead to do the HTML parsing.

I think it is OK for now, because this is how the OpenPrescribing code works (but it could also be refactored there in future).

I'd consider opening a deck scrubbing issue to revisit switching to lxml.

Reply:
My first draft of this used lxml, but I switched to the prior art implementation from OpenPrescribing as it has seen a lot of use in production and thus can be relied on